### PR TITLE
[SPARK-36303][SQL]  Label error classes for subset of QueryExecutionErrors 

### PR DESCRIPTION
**Pull request by dgd-contributor**
### What changes were proposed in this pull request?
Adds error classes to some of the exceptions in QueryExecutionErrors.

### Does this PR introduce _any_ user-facing change?
Fills in missing error class and SQLSTATE fields.

### How was this patch tested?
Existing tests

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @karenfeng FYI

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1,8 +1,61 @@
 {
+  "ADD_FILES_WITH_ABSOLUTE_PATH_UNSUPPORTED_ERROR" : {
```
Comment (karenfeng) [inline core/src/main/resources/error/error-classes.json:2]: Let's remove the `_ERROR` boilerplate suffix for simplicity.

File: core/src/main/scala/org/apache/spark/SparkException.scala [hunk 2]
```diff
@@ -79,3 +83,66 @@ class SparkArithmeticException(errorClass: String, messageParameters: Array[Stri
...
+  extends UnsupportedOperationException(
+    SparkThrowableHelper.getMessage(errorClass, messageParameters))
+    with SparkThrowable {
+
+  override def getErrorClass: String = errorClass
+  override def getSqlState: String = SparkThrowableHelper.getSqlState(errorClass)
+}
+
+/**
+ * File doesn't be found exception thrown from Spark with an error class.
```
Comment (karenfeng) [inline core/src/main/scala/org/apache/spark/SparkException.scala:100]: Nit: doesn't -> not

File: core/src/main/scala/org/apache/spark/SparkException.scala [hunk 3]
```diff
@@ -79,3 +83,66 @@ class SparkArithmeticException(errorClass: String, messageParameters: Array[Stri
...
+  extends RuntimeException(
+    SparkThrowableHelper.getMessage(errorClass, messageParameters))
+    with SparkThrowable {
+
+  override def getErrorClass: String = errorClass
+  override def getSqlState: String = SparkThrowableHelper.getSqlState(errorClass)
+}
+
+/**
+ * NoSuch element exception thrown form Spark with an error class.
```
Comment (karenfeng) [inline core/src/main/scala/org/apache/spark/SparkException.scala:139]: Nit: NoSuch -> No such

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 4]
```diff
@@ -1406,111 +1406,144 @@ object QueryExecutionErrors {
...
-      s"'$name': $pluginClassName", e.getCause)
+    new SparkException(
+      errorClass = "FAILED_TO_INSTANTIATE_CONSTRUCTOR_FOR_CATALOG_ERROR",
+      messageParameters = Array(name, pluginClassName), e.getCause)
   }
 
   def noSuchElementExceptionError(): Throwable = {
-    new NoSuchElementException
+    new SparkNoSuchElementException(
+      errorClass = "NO_SUCH_ELEMENT_EXCEPTION_ERROR",
```
Comment (karenfeng) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1519]: We can further simplify this error class to `NO_SUCH_ELEMENT`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 5]
```diff
@@ -1406,111 +1406,144 @@ object QueryExecutionErrors {
...
+      messageParameters = Array.empty
+    )
   }
 
   def noSuchElementExceptionError(key: String): Throwable = {
     new NoSuchElementException(key)
```
Comment (karenfeng) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:1525]: Can you also fix this one? Should be straightforward; we can use the same errorClass as `noSuchElementExceptionError()` but change it to have `"message": [ "%s" ]`, and pass the key here in `messageParameters`.

**Review by karenfeng — COMMENTED** [review]
Thanks for working on this @dgd-contributor! I left some high-level comments. Can you update the SQLSTATEs according to the error class [README](https://github.com/apache/spark/blob/master/core/src/main/resources/error/README.md)? 42000 refers only to semantic and syntax errors.

**Pull request closed without merging**